src/security/Handshake.cc

```diff
@@ -504,6 +506,27 @@ Security::HandshakeParser::parseSniExtension(const SBuf &extensionData) const
     return SBuf(); // SNI extension lacks host_name
 }
 
+// RFC 8446 Section 4.2.1: SupportedVersions extension
+void
+Security::HandshakeParser::parseSupportedVersionsExtension(const SBuf &extensionData) const
+{
+    if (handshakeType == hskClientHello) {
+        Parser::BinaryTokenizer tkList(extensionData);
+        Parser::BinaryTokenizer tkVersions(tkList.pstring8("SupportedVersionsList"));
+        while (!tkVersions.atEnd()) {
+            Parser::BinaryTokenizerContext version(tkVersions, "SupportedVersion");
+            const uint16_t aVersion = tkVersions.uint16(".version");
+            if (aVersion == 0x0304)
+                details->tlsSupportedVersion = AnyP::ProtocolVersion(AnyP::PROTO_TLS, 1, 3);
```

Comment:
I used a different form now to break the while loop. Not sure which is the best (because in the future we may need to store all listed versions, anyway not important).
Please check commit 98e8600f.